### PR TITLE
metrics: pass criterion

### PR DESCRIPTION
**Pull request by casperdcl**

### CODE REVIEW

File: SIRF_data_preparation/evaluation_utilities.py [hunk 1]
```diff
@@ -15,16 +15,31 @@ def read_objectives(datadir='.'):
...
+    Returns first index of `metrics` with value <= `thresh`.
+    The value must remain below the threshold for at least `window`.
+    Raises IndexError if doesn't pass.
+    """
+    assert metrics.shape[1] == len(thresh)
+    assert metrics.ndim == 2
+    assert thresh.ndim == 1
+
+    m = (metrics <= thresh[None]).all(1)
+    assert window == 1
```
Comment (KrisThielemans) [inline SIRF_data_preparation/evaluation_utilities.py:38]: I guess this is a future feature? Fine for me

Comment (casperdcl) [reply SIRF_data_preparation/evaluation_utilities.py]: Added implementation.

File: SIRF_data_preparation/evaluation_utilities.py [hunk 2]
```diff
@@ -15,16 +15,34 @@ def read_objectives(datadir='.'):
...
-    return numpy.asarray([
+    return np.asarray([
         list(qm.evaluate(STIR.ImageData(str(Path(srcdir) / f'iter_{i:04d}.hv'))).values()) for i in iters])
 
 
-def plot_metrics(iters: Iterator[int], m: numpy.ndarray, labels=None, suffix=""):
+def pass_index(metrics: np.ndarray, thresh: np.ndarray, window: int = 1) -> int:
+    """
+    Returns first index of `metrics` with value <= `thresh`.
+    The value must remain below the threshold for at least `window`.
```
Comment (KrisThielemans) [inline SIRF_data_preparation/evaluation_utilities.py:30]:
```suggestion
    The values must remain below the respective thresholds for at least `window` number of entries.
```

**Review by KrisThielemans — APPROVED** [review]
possibly accept list for `thresh`, although it doesn't really matter. Otherwise, it seems to work fine!

**Pull request merged**